src/utils/vvim.cpp

```diff
@@ -2047,12 +2056,26 @@ bool VVim::handleKeyPressEvent(int key, int modifiers, int *p_autoIndentPos)
                 addRangeToken(range);
                 processCommand(m_tokens);
                 break;
-            } else if (hasActionToken() || !checkMode(VimMode::Normal)) {
-                // Invalid sequence.
-                break;
-            } else if (checkPendingKey(Key(Qt::Key_BracketLeft))) {
+            } else if (hasActionToken() && checkPendingKey(Key(Qt::Key_BracketLeft))) {
                 // [{, goto previous title at one higher level.
                 processTitleJump(m_tokens, false, -1);
+            } else if (checkMode(VimMode::Normal) || checkMode(VimMode::Visual) || checkMode(VimMode::VisualLine)) {
```

Comment:
The Vim mode implementation is a little complicated in VNote, since we need to take of the `[count]` and other stuff. In VNote, we abstract the Vim operations like `[repeat][action][movement]`. For example, `2{`, `d2{`, `c{`, or `y{` is not considered and handled here. You could reference other movement's implementations for the details.

Reply:
Got it. Thanks for your suggestion, it's helpful!